src/CustomIndexing/CustomIndexing.jl

```diff
@@ -0,0 +1,130 @@
+module CustomIndexing
+
+"""
+    struct SparseArray{T,N} <: AbstractArray{T,N}
```

Comment:
Does it really make sense to call this an `AbstractArray` if `size` and `axes` aren't implemented?

Reply:
If it is not an `AbstractArray` then `map(f, d)` will store the result in a `Vector` and it's difficult to make it behave differently except by being very hacky with `Base._..` functions